### PR TITLE
Implement DeltaSharingClient.getCDFFiles

### PR DESCRIPTION
**Pull request by linzhou-db**
- Implement DeltaSharingClient.getCDFFiles
- Extends Metadata with configuration
- Exposes DeltaStandaloneException from the OSS Delta Sharing Server as client error

### CODE REVIEW

File: spark/src/main/scala/io/delta/sharing/spark/model.scala
Comment (chakankardb) [inline spark/src/main/scala/io/delta/sharing/spark/model.scala:100]: There is some conflict here with changes in my PR:
https://github.com/delta-io/delta-sharing/pull/149

Perhaps wait for that PR before we merge this one?

Comment (linzhou-db) [reply spark/src/main/scala/io/delta/sharing/spark/model.scala:100]: sg, will do

**Review by chakankardb — APPROVED** [review]

File: server/src/main/scala/io/delta/sharing/server/DeltaSharingService.scala
Comment (chakankardb) [inline server/src/main/scala/io/delta/sharing/server/DeltaSharingService.scala:100]: Is it safe to always map delta standalone exceptions to be bad requests?

Comment (linzhou-db) [reply server/src/main/scala/io/delta/sharing/server/DeltaSharingService.scala:100]: In this case it is a bad request instead of an internal error. If we do not expose all `DeltaStandaloneException`, then we have to do a special match for this case, which also works, but don't feel it's obviously better than the current solution.

Will quickly sync with Ryan tomorrow to get more context.

Comment (chakankardb) [reply server/src/main/scala/io/delta/sharing/server/DeltaSharingService.scala:100]: I added a more specific check in: https://github.com/delta-io/delta-sharing/pull/150

**Pull request merged**
